File: src/main/java/com/HOIIVUtils/clauzewitz/script/IntegerPDX.java
package com.HOIIVUtils.clauzewitz.script;

import com.HOIIVUtils.clausewitz_parser.Node;
import com.HOIIVUtils.clausewitz_parser.NodeValue;

import java.util.List;

public class IntegerPDX extends AbstractPDX<Integer> {
    public IntegerPDX(String pdxIdentifiers) {
        super(pdxIdentifiers);
    }

    public IntegerPDX(String... PDXIdentifiers) {
        super(PDXIdentifiers);
    }

    public IntegerPDX(List<String> pdxIdentifiers) {
        super(pdxIdentifiers);
    }

    @Override
    public void set(Node expression) throws UnexpectedIdentifierException, NodeValueTypeException {
        usingIdentifier(expression);
        NodeValue value = expression.value();

        if (value.valueObject() instanceof Number num) {
            obj = num.intValue();
        } else {
            throw new NodeValueTypeException(expression, "Number (as an Integer)");
        }
    }

    @Override
    public boolean objEquals(PDXScript<?> other) {
        if (other instanceof IntegerPDX pdx) {
            return obj.equals(pdx.get());
        }
        return false;
    }
}

File: src/main/scala/com/hoi4utils/clausewitz/script/IntegerPDX.scala
package com.hoi4utils.clausewitz.script

import com.hoi4utils.clausewitz_parser.Node
import com.hoi4utils.clausewitz_parser.NodeValue

import java.util
import java.util.List


class IntegerPDX extends AbstractPDX[Integer](pdxIdentifiers) {
  def this(pdxIdentifiers: String) = {
    this()
  }

  def this(PDXIdentifiers: String*) = {
    this()
  }

  def this(pdxIdentifiers: util.List[String]) = {
    this()
  }

  @throws[UnexpectedIdentifierException]
  @throws[NodeValueTypeException]
  override def set(expression: Node): Unit = {
    usingIdentifier(expression)
    this.node = expression
    if (!valueIsInstanceOf(Int)) {
      throw new NodeValueTypeException(expression, "Number (as an Integer)")
    }
  }

  override def nodeEquals(other: PDXScript[_]): Boolean = {
    if (other.isInstanceOf[IntegerPDX]) return node.equals(other.getNode)
    false
  }
}
